src/main/java/de/srendi/advancedperipherals/common/util/inventory/ItemFilter.java

```diff
@@ -95,19 +102,35 @@ public static Pair<ItemFilter, String> parse(Map<?, ?> item) {
     }
 
     public static ItemFilter fromStack(ItemStack stack) {
-        ItemFilter filter = empty();
+        ItemFilter filter = createEmpty();
         filter.item = stack.getItem();
         filter.componentsAsNbt = DataComponentUtil.toNbt(stack.getComponentsPatch());
         filter.components = (PatchedDataComponentMap) stack.getComponents();
         return filter;
     }
 
-    public static ItemFilter empty() {
+    public static ItemFilter createEmpty() {
         return new ItemFilter();
     }
 
     public boolean isEmpty() {
-        return fingerprint.isEmpty() && item == Items.AIR && tag == null && componentsAsNbt == null;
+        return this == EMPTY || (fingerprint.isEmpty() && item == Items.AIR && tag == null && componentsAsNbt == null);
+    }
+
+    @Override
+    public boolean testAE(GenericStack genericStack) {
```

Comment:
have you tested if these functions can be load without AE or RS installed?

Reply:
you are right
Had it in mind to test that later, but I forgot to